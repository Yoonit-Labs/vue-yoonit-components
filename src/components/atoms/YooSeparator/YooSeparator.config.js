module.exports = {
  fill: {
    options: [
      'default',
      'primary',
      'danger'
    ]
  }
}
