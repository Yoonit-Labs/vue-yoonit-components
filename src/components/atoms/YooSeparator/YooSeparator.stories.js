import yooSeparator from './YooSeparator.vue'
import PropsConfig from './YooSeparator.config'

export default {
  title: 'Yoonit/YooSeparator',
  component: yooSeparator,
  argTypes: {
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooSeparator },
  template:
    '<yoo-separator v-bind="$props"></yoo-separator>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Warning test component' }

export const Default = Template.bind({})
Default.args = { text: 'Yoo Warning test component', fill: 'default' }

export const Primary = Template.bind({})
Primary.args = { text: 'Yoo Warning test component', fill: 'primary' }

export const Danger = Template.bind({})
Danger.args = { text: 'Yoo Warning test component', fill: 'danger' }
