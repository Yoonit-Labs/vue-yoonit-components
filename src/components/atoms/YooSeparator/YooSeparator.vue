<template lang="pug">
div
  div(
    :class="['separator', ...takeModifier]"
  )
    p(
      class="separator__label m__l--l m__t--none m__b--none m__r--none"
      v-show="text"
    )
      | {{text}}
</template>

<script>
export default {
  name: 'YooSeparator',
  props: {
    fill: {
      type: String,
      default: 'default',
      validator (value) {
        return [
          'default',
          'primary',
          'danger'
        ].indexOf(value) !== -1
      }
    },
    text: {
      type: String,
      required: false,
      validator: (value) => {
        return typeof value === 'string'
      },
      default: ''
    }
  },
  data: () => ({
  }),
  created () {},
  mounted () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  components: {},
  computed: {
    /**
    * @description Print classes based on the chosen props
    * @computed takeModifier
    * @returns {array}
    */
    takeModifier () {
      const block = 'separator'
      const classList = []

      if (this.fill) {
        classList
          .push(
            `${block}--fill-${this.fill}`
          )
      }

      return classList
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooSeparator.sass" lang="sass" scoped></style>
