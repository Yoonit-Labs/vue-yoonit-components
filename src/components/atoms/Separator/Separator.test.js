import { shallowMount } from '@vue/test-utils'
import YooSeparator from '@/components/atoms/Separator/Separator.vue'
import PropsConfig from '@/components/atoms/Separator/Separator.config'

const classBlock = 'yoo-separator'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooSeparator, {
    slots: { default: SlotText }
  })
}

describe('YooSeparator Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-separator').exists()).toBe(true)
    expect(wrapper.find('.yoo-separator__label').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('fill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fill.options.includes(YooSeparator.props.fill.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(fill => {
        it(`Includes variation class: .${classBlock}__fill--${fill}`, async () => {
          await wrapper.setProps({ fill })
          expect(wrapper.find(`.${classBlock}__fill--${fill}`).exists()).toBe(true)
        })
      })
    })

    describe('fontSize', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fontSize.options.includes(YooSeparator.props.fontSize.default)).toBe(true)
      })
      PropsConfig.fontSize.options.forEach(fontSize => {
        it(`Includes variation class: .${classBlock}__font--${fontSize}`, async () => {
          await wrapper.setProps({ fontSize })
          expect(wrapper.find(`.${classBlock}__font--${fontSize}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
})
