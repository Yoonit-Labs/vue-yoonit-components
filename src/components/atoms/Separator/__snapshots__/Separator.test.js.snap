// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooSeparator Component Matches Snapshot 1`] = `
<div>
  <div class="yoo-separator yoo-separator__fill--default yoo-separator__font--md">
    <p class="m__l--m m__t--none m__b--none m__r--none yoo-separator__label yoo-separator__fill--default yoo-separator__font--md" style="display: none;"></p>
  </div>
</div>
`;
